extensions/wikia/ContentReview/tests/ContentReviewHelperTest.php

```diff
@@ -8,61 +8,119 @@ public function setUp() {
+		$this->assertEquals( $expected, $value );
+	}
+
+	/**
 	 * Test for \Wikia\ContentReview\Hooks::onRawPageViewBeforeOutput hook
+	 *
+	 * To review different cases - check the documentation of the dataProvider.
+	 *
+	 * @dataProvider replaceWithLastApprovedRevisionProvider
+	 * @param bool $isPageReviewed
+	 * @param int $latestRevID
+	 * @param array $latestReviewedRevision
+	 * @param bool $isTestModeEnabled
+	 * @param bool $revisionExists
+	 * @param string $originalText
+	 * @param string $lastReviewedText
+	 * @param string $expectedText
 	 */
-	public function testReplaceWithLastApprovedRevision( $params, $textExpected, $message ) {
-
-		/* @var \Title $titleMock */
-		$titleMock = $this->getMockBuilder( '\Title' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'getArticleID', 'getLatestRevID', 'isJsPage' ] )
-			->getMock();
-		$titleMock->method( 'getArticleID' )
-			->will( $this->returnValue( $params['pageId'] ) );
-		$titleMock->method( 'getLatestRevID' )
-			->will( $this->returnValue( $params['latestRevId'] ) );
-		$titleMock->method( 'isJsPage' )
-			->will( $this->returnValue( $params['isJsPage'] ) );
-
-		/* @var \Wikia\ContentReview\Models\CurrentRevisionModel $currentRevisionModelMock */
-		$currentRevisionModelMock = $this->getMockBuilder( '\Wikia\ContentReview\Models\CurrentRevisionModel' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'getLatestReviewedRevision' ] )
-			->getMock();
-		$currentRevisionModelMock->method( 'getLatestReviewedRevision' )
-			->will( $this->returnValue( $params['latestApprovedRevData'] ) );
-
-		/* @var \Revision $revisionMock */
-		$revisionMock = $this->getMockBuilder( '\Revision' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'getRawText' ] )
-			->getMock();
-		$revisionMock->method( 'getRawText' )
-			->will( $this->returnValue( $params['latestApprovedRevText'] ) );
-
-		$this->mockGlobalVariable( 'wgJsMimeType', $params['wgJsMimeType'] );
-
-		/* @var \Wikia\ContentReview\Helper $helperMock */
-		$helperMock = $this->getMockBuilder( '\Wikia\ContentReview\Helper' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'getCurrentRevisionModel', 'getRevisionById', 'isContentReviewTestModeEnabled' ] )
-			->getMock();
-		$helperMock->method( 'getCurrentRevisionModel' )
-			->will( $this->returnValue( $currentRevisionModelMock ) );
-		$helperMock->method( 'isContentReviewTestModeEnabled' )
-			->will( $this->returnValue( $params['isContentReviewTestModeEnabled'] ) );
-		$helperMock->method( 'getRevisionById' )
-			->will( $this->returnValue( $revisionMock ) );
-
-		$helperMock->replaceWithLastApproved( $titleMock, $params['contentType'], $params['text'] );
-
-		$this->assertEquals( $textExpected, $params['text'], $message );
+	public function testReplaceWithLastApprovedRevision( $isPageReviewed, $latestRevID, array $latestReviewedRevision,
+		$isTestModeEnabled, $revisionExists, $originalText, $lastReviewedText, $expectedText, $message
```

Comment:
This method signature is getting a bit big. I'd suggest switching to passing an array of test data, which will also make your data provider clearer (without the need to comment array entries) since the keys will be descriptive (i.e. https://github.com/Wikia/app/blob/3efa2cfe7144f/extensions/wikia/ProtectSiteJS/tests/ProtectSiteJSTest.php#L49-L56).